.github/workflows/docker.yaml

```diff
@@ -38,27 +38,24 @@ jobs:
             type=sha,prefix=,format=long
             type=semver,pattern=v{{version}}
             type=semver,pattern=v{{major}}.{{minor}}
-      
-      - name: Don't push to registry if this is a PR
-        if: "${{ github.event_name }}" = "pull_request"
-        run: |
-          echo "Not pushing the image to any container registry as this workflow is running on a pull request"
-          exit 0
 
       - name: Log in to DockerHub
+        if: github.event_name != 'pull_request'
```

Comment:
We should explain something about why here. This allows us to see if the Docker build is failing without cluttering our list of Docker images (or something)

Reply:
It's more that we don't want to upload docker images of in-progress PRs (without doing so manually via `workflow_dispatch`).

There's an explanation above by `pull_request:`. I've extended it with *why* in e2b591a3f9ac35924f790e65f1f250a3672cf7ff.